sockeye/train.py

```diff
@@ -79,14 +87,41 @@ def main():
     assert args.optimized_metric == C.BLEU or args.optimized_metric in args.metrics, \
         "Must optimize either BLEU or one of tracked metrics (--metrics)"
 
+    # Checking status of output folder, resumption, etc.
     output_folder = os.path.abspath(args.output)
-    if not os.path.exists(output_folder):
+    resume_training = False
+    training_state_dir = os.path.join(output_folder, C.TRAINING_STATE_DIRNAME)
+    if os.path.exists(output_folder):
+        if args.overwrite_output:
+            shutil.rmtree(output_folder)
+            os.makedirs(output_folder)
+        elif os.path.exists(training_state_dir):
+            with open(os.path.join(output_folder, C.ARGS_STATE_NAME), "r") as fp:
+                old_args = json.load(fp)
+            arg_diffs = _dict_difference(vars(args), old_args) | _dict_difference(old_args, vars(args))
+            # Remove args that may differ without affecting the training.
+            # TODO: should we include device-ids and use-cpu?
+            # Are there more?
+            arg_diffs -= set(["overwrite_output", "use-tensorboard", "quiet", "align_plot_prefix", "sure_align_threshold"])
+            if not arg_diffs:
+                resume_training = True
+            else:
+                # We do not have the logger yet
```

Comment:
I see, 

can we create a logger two times?
once before this block without a path argument, and then with path as before?